lib/onebox/engine/gfycat_onebox.rb

```diff
@@ -62,27 +58,50 @@ def match
         @match ||= @url.match(/^https?:\/\/gfycat\.com\/(gifs\/detail\/)?(?<name>.+)/)
       end
 
+      def nokogiri_page
+        @nokogiri_page ||= begin
+          response = Onebox::Helpers.fetch_response(url, 10) rescue nil
+          Nokogiri::HTML(response)
+        end
+      end
+
+      def get_og_data
+        og_data = {}
+
+        if json_string = nokogiri_page.at_css('script[type="application/ld+json"]')&.text
+          og_data = Onebox::Helpers.symbolize_keys(::MultiJson.load(json_string))
+        end
+
+        og_data
+      end
+
       def data
+        og_data = get_og_data
 
-        total_tags = [raw['gfyItem']['tags'], raw['gfyItem']['userTags']].flatten.compact
-        tag_links = total_tags.map { |t| "<a href='https://gfycat.com/gifs/search/#{t}'>##{t}</a>" }.join(' ') if total_tags
-        autoplay = raw['gfyItem']['webmSize'].to_i < 10485760 ? 'autoplay' : ''
-
-        {
-          name: raw['gfyItem']['gfyName'],
-          title: raw['gfyItem']['title'] || 'No Title',
-          author: raw['gfyItem']['userName'],
-          tags: tag_links,
-          url: @url,
-          posterUrl: raw['gfyItem']['posterUrl'],
-          webmUrl: raw['gfyItem']['webmUrl'],
-          mp4Url: raw['gfyItem']['mp4Url'],
-          width: raw['gfyItem']['width'],
-          height: raw['gfyItem']['height'],
-          autoplay: autoplay
+        response = {
+          name: match[:name],
+          title: og_data[:headline] || 'No Title',
+          author: og_data[:author],
+          url: @url
         }
-      end
 
+        keywords = og_data[:keywords]&.split(',')
+        if keywords
+          response[:keywords] = keywords.map { |t| "<a href='https://gfycat.com/gifs/search/#{t}'>##{t}</a>" }.join(' ')
+        end
+
+        if og_data[:video]
+          video_url = Pathname.new(og_data[:video][:contentUrl])
```

Comment:
I see that we were not normalizing the URL previously in gfycat oneboxes but as a precaution we should remove unsafe characters from URL using `::Onebox::Helpers.normalize_url_for_output(og_data[:video][:contentUrl])`.

Same applies to `og_data[:video][:thumbnailUrl]` below.

Reply:
Thx, applied modifications and merged it!